newIDE/app/src/ResourcesList/ResourceSelector.js

```diff
@@ -169,6 +170,7 @@ export default class ResourceSelector extends React.Component<Props, State> {
       () => {
         if (!this.state.notExistingError) {
           if (this.props.onChange) this.props.onChange(resourceName);
+          if (this.props.onRequestClose) this.props.onRequestClose();
```

Comment:
The issue regarding ResourceSelector I believe was not having enough time to properly select the option. The autocomplete was closing even before changing anything. I could've maybe used a timeout for calling onRequestClose, but I think this is a more rigid and proper way, as it would ensure that the necessary changes have been done before closing the popup. The function is not called from autocomplete, rather it is being called from the ResourceSelector itself.

Reply:
I think I know why: 
https://github.com/4ian/GDevelop/blob/a6ae265705fa6d12bb8d05007dfd446c1e8f782a/newIDE/app/src/ResourcesList/ResourceSelector.js#L159-L175

The `this.props.onChange` is only called in the *callback* of setState. Meaning that if the component is blurred and destroyed, it won't be called, because the component will be unmounted by this time.

We can try to fix that in another PR as this can be tricky.

Reply:
> Meaning that if the component is blurred and destroyed, it won't be called, because the component will be unmounted by this time.

This was the exact behaviour, no change happened because the component unmounted by this time. I've called the `onRequestClose` in the callback of setState, after `onChange`. Will it bring more errors? (If yes, then I'd drop it for this PR, and work on something optimal and proper in another PR :) )

Reply:
> I've called the onRequestClose in the callback of setState, after onChange. Will it bring more errors

In your PR? I don't see any onRequestClose in a callback of a setState?
Calling it after onChange is ok, because onChange will almost never result in closing a dialog - and even if it's the case, the important is to not call anything related to the component in a **callback** (setTimeout/setState callback).
Calling `onRequestClose` after `onChange` (or `onChoose`) is fine, as it's onRequestClose that will trigger the closing (well, probably ;)).

Right now I think it's ResourceSelector that is "buggy", because `onChange` is called in the callback of setState - while it's both:

* not necessary
* making the onChange not work because in the meantime, the component can have been unmounted.

Let's try to fix ResourcesSelector in another PR.